Add Cohere docs (#108)
strands-agents/docs: docs/user-guide/concepts/model-providers/cohere.md

# Cohere

[Cohere](https://cohere.com) provides cutting-edge language models. These are accessible accessible through OpenAI's SDK via the Compatibility API. This allows easy and portable integration with the Strands Agents SDK using the familiar OpenAI interface.

## Installation

> Cohere is configured as an optional dependency in Strands Agents via the openai sdk. To install, run:

Reviewer: Can we update this to:
```suggestion
The Strands Agents SDK provides access to Cohere models through the OpenAI compatiblity layer, configured as an optional dependency. To install, run:
```

```bash
pip install 'strands-agents[openai]'

> ````

Reviewer: One too many backticks is causing a rendering issue

## Usage

After installing the `openai` package, you can import and initialize the Strands Agents' OpenAI-compatible provider for Cohere models as follows:

```python
from strands import Agent
from strands.models.openai import OpenAIModel
from strands_tools import calculator

model = OpenAIModel(
    client_args={
        "api_key": "<COHERE_API_KEY>",
        "base_url": "https://api.cohere.ai/compatibility/v1",  # Cohere compatibility endpoint
    },

>     model_id="command-a-03-2025"  # or see https://docs.cohere.com/docs/models

Reviewer: Right now, we need the following additional parameter to use the cohere compatibility endpoint:

```suggestion
    model_id="command-a-03-2025"  # or see https://docs.cohere.com/docs/models,
    params={
        "stream_options": None
    }
```

)

agent = Agent(model=model, tools=[calculator])

> response = agent("What is 2+2?")
> print(response)

Reviewer: No need to print, this prints by default:
```suggestion
agent("What is 2+2?")
```

```

## Configuration

### Client Configuration

The `client_args` configure the underlying OpenAI-compatible client. When using Cohere, you must set:

* `api_key`: Your Cohere API key. Get one from the [Cohere Dashboard](https://dashboard.cohere.com).

> * `base_url`: `https://api.cohere.ai/compatibility/v1`

Reviewer: nit: Can you move the url to a bullet point on a new line, It makes it much easier to copy paste with a double click

Author: ```suggestion
* `base_url`:
    * `https://api.cohere.ai/compatibility/v1`
```

# Cohere

[Cohere](https://cohere.com) provides cutting-edge language models. These are accessible accessible through OpenAI's SDK via the Compatibility API. This allows easy and portable integration with the Strands Agents SDK using the familiar OpenAI interface.

## Installation

The Strands Agents SDK provides access to Cohere models through the OpenAI compatiblity layer, configured as an optional dependency. To install, run:

```bash
pip install 'strands-agents[openai]'
```

## Usage

After installing the `openai` package, you can import and initialize the Strands Agents' OpenAI-compatible provider for Cohere models as follows:

```python
from strands import Agent
from strands.models.openai import OpenAIModel
from strands_tools import calculator

model = OpenAIModel(
    client_args={
        "api_key": "<COHERE_API_KEY>",
        "base_url": "https://api.cohere.ai/compatibility/v1",  # Cohere compatibility endpoint
    },

>     model_id="command-a-03-2025"  # or see https://docs.cohere.com/docs/models,

Reviewer 2: The comma is commented out; it needs to be before the #.

`model_id="command-a-03-2025" # or see https://docs.cohere.com/docs/models,` -> `model_id="command-a-03-2025", # or see https://docs.cohere.com/docs/models`
